src/map/roe.cpp

```diff
@@ -21,20 +21,31 @@
 
 ===========================================================================
 */
+#include <time.h>
 
 #include "roe.h"
+#include "vana_time.h"
 #include "../common/lua/lunar.h"
 #include "lua/luautils.h"
 #include "packets/chat_message.h"
 #include "utils/charutils.h"
+#include "utils/zoneutils.h"
 
 #include "packets/roe_questlog.h"
 #include "packets/roe_sparkupdate.h"
 #include "packets/roe_update.h"
 
+#define ROE_CACHETIME 15
+
 std::array<RoeCheckHandler, ROE_NONE> RoeHandlers;
 RoeSystemData roeutils::RoeSystem;
 
+void SaveEminenceDataNice(CCharEntity* PChar)
+{
+    if (PChar->m_eminenceCache.lastWriteout < time(nullptr) - ROE_CACHETIME)
+        charutils::SaveEminenceData(PChar);
```

Comment:
This has been codified in the style guide:
https://github.com/project-topaz/topaz/blob/release/CONTRIBUTING.md#c

`-checks="readability-braces-around-statements"`
```cpp
// Correct ✔️ 
if (x == 5)
{
    function();
}

// Wrong ❌
if (x == 5)
    function();

// Wrong ❌
if (x == 5)
    function(21);
else
{
    function(42);
}
```